### DISCUSSION TITLE
Using RAG when running on Unraid?

### DISCUSSION THREAD

**Discussion by Jayphen** [danny-avila/LibreChat#4312] [Help Wanted]
When adding the RAG_API_URL and the RAG_OPENAI_API_KEY to the env variables as per the docs, the RAG API does not seem to work as intended. I get the following error in the logs:

```
warn: RAG API is either not running or not reachable at http://host.docker.internal:8000, you may experience errors with file uploads.
```

Is there additional setup required to have this work inside unraid?

Thanks!

**Comment by volcs0** [comment]
Did you solve this issue? I'm having the exact issue and can't get file uploads to work. Thanks.

**Reply by Jayphen** [reply]
I didn't solve it. I was hoping someone else could help :D

**Reply by trytofly94** [reply]
Any update? Is RAG just impossible with Librechat on Unraid?

**Comment by trytofly94** [comment]
Has there been any update on this issue?

**Comment by EvilPudding** [comment]
Did you add those env variables to the rag service or librechat? `http://host.docker.internal:8000` is just an example, you're meant to replace it with the actual address of the rag service you're hosting.
The docs show all steps required in running the service, but unfortunately, up to this date, there is no unraid community app config for the librechat's rag api service.

**Comment by JonReed** [comment]
this is an old discussion, but commenting since I came across it with no solution

in the Extra Parameters box for the config of the unraid docker (enable advanced to see) add:

`--add-host=host.docker.internal:host-gateway`

then http://host.docker.internal:8000 worked for me (I have a follow on problem, but this has let me progress)
